packages/s/sfml/xmake.lua

```diff
@@ -122,115 +120,77 @@ package("sfml")
-                    package:add("syslinks", "opengl32", "gdi32", "user32", "advapi32")
-                end
-                if package:is_plat("linux") then
-                    package:add("deps", "libx11", "libxext", "libxrandr", "libxrender", "freetype", "eudev")
-                    package:add("deps", "opengl", "glx", {optional = true})
-                end
+                package:add("deps", "libx11", "libxcursor", "libxext", "libxrandr", "libxrender", "eudev")
+                package:add("deps", "opengl", "glx", {optional = true})
             end
-            if package:config("audio") then
-                package:add("links", a .. "audio" .. e)
-                if package:is_plat("windows", "mingw") and not package:config("shared") then
-                    package:add("links", "openal32", "flac", "vorbisenc", "vorbisfile", "vorbis", "ogg")
-                elseif package:is_plat("linux") then
-                    package:add("deps", "libogg", "libflac", "libvorbis", "openal-soft")
-                end
-            end
-            if package:config("network") then
-                package:add("links", a .. "network" .. e)
-                if package:is_plat("windows", "mingw") and not package:config("shared") then
-                    package:add("syslinks", "ws2_32")
-                end
-            end
-            if package:is_plat("windows", "mingw") and package:config("main") then
-                package:add("links", main_module)
-            end
-            package:add("links", a .. "system" .. e)
-            if package:is_plat("windows", "mingw") then
-                package:add("syslinks", "winmm")
+        end
+
+        if package:config("audio") then
+            package:add("deps", "libogg", "libflac", "libvorbis", "openal-soft")
+        end
+
+        package:add("components", "system")
+        for _, component in ipairs({"graphics", "window", "audio", "network"}) do
+            if package:config(component) then
+                package:add("components", component)
             end
         end
+
+        if package:is_plat("windows", "mingw") and package:config("main") then
+            package:add("components", "main")
+        end
     end)
 
-    on_install("windows", "linux", function (package)
+    on_install("windows", "linux", "macosx", "mingw", function (package)
         local configs = {"-DSFML_BUILD_DOC=OFF", "-DSFML_BUILD_EXAMPLES=OFF"}
         table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
         if package:config("shared") then
             table.insert(configs, "-DBUILD_SHARED_LIBS=ON")
+            -- Fix missing system libs
+            if package:is_plat("windows", "mingw") then
+                local file = io.open("src/SFML/Audio/CMakeLists.txt", "a")
+                file:print("target_link_libraries(OpenAL PRIVATE winmm)")
+                file:close()
```

Comment:
I tried it, winmm will still be added to the front of the openal library link in linksLib.rsp, causing the link to fail.

Reply:
Are you sure you tried the same thing? You posted a code adding the lib to the `sfml-audio` target but here I'm adding it to the `OpenAL` target (which is implicitly created by SFML).

Reply:
and target name is sfml-audio instead of OpenAL

Reply:
> Are you sure you tried the same thing? You posted a code adding the lib to the `sfml-audio` target but here I'm adding it to the `OpenAL` target (which is implicitly created by SFML).

oh, it works, but we should use `INTERFACE` instead of `PRIVATE `

Reply:
`OpenAL` is a target created by the sfml_add_external macro defined here: <https://github.com/SFML/SFML/blob/2.6.x/cmake/Macros.cmake#L328>

which is probably why the flag order is messed up

Reply:
and we need also patch graphics

```
          if package:is_plat("windows", "mingw") and package:config("shared") then
                local file = io.open("src/SFML/Graphics/CMakeLists.txt", "a")
                file:print("target_link_libraries(Freetype INTERFACE z)")
                file:close()
            end
  ```

Reply:
Freetype can depends on other libs, depending on the options enabled for it, we should add every dependent lib name too (and probably link dir as well), like packagedeps does but manually